### PR TITLE
Fix issue with expiring repository certificates on Azure on RHEL 7

### PR DESCRIPTION
**Pull request by erzetpe**

### CODE REVIEW

File: ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml [hunk 1]
```diff
@@ -1,4 +1,16 @@
 ---
+- include_vars:
+    file: roles/common/vars/main.yml
+    name: common_vars
+
+# Fix to solve issues with expiring certificates on Azure
+# https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/redhat/redhat-rhui#update-expired-rhui-client-certificate-on-a-vm
+# TODO: Needs to be updated for RHEL 8
+- name: Update repository certificates
+  command: yum update -y --disablerepo='*' --enablerepo='*microsoft*'
```
Comment (atsikham) [inline ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml]: Change to yum module - https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_module.html#parameter-update_only

File: ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+  when:
+    - common_vars.provider == "azure" and ansible_distribution_major_version == "7"
```
Comment (atsikham) [inline ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml]: add `ansible_os_family` check, use the list of conditions:

```yaml
when:
 - condition0
 - condition1
 - condition2
```

**Review by atsikham — CHANGES_REQUESTED** [review]

File: ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml (hunk 1 above)
Comment (erzetpe) [reply ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml]: Done.

File: ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml (hunk 2 above)
Comment (erzetpe) [reply ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml]: Done.

File: ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml [hunk 3]
```diff
@@ -1,4 +1,22 @@
...
+# TODO: Needs to be updated for RHEL 8
+- name: Update repository certificates
+  yum:
+    enablerepo: "*microsoft*"
+    disablerepo: "*"
+    state: "latest"
+    name: '*'
+  when:
+    - common_vars.provider == "azure"
+    - ansible_os_family == "RedHat"
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml]:
```suggestion
    - ansible_distribution == "RedHat"
```
This is to skip this task on CentOS/AlmaLinux

File: ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml [hunk 4]
```diff
@@ -1,4 +1,22 @@
...
+
+# Fix to solve issues with expiring certificates on Azure
+# https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/redhat/redhat-rhui#update-expired-rhui-client-certificate-on-a-vm
+# TODO: Needs to be updated for RHEL 8
+- name: Update repository certificates
+  yum:
+    enablerepo: "*microsoft*"
+    disablerepo: "*"
+    state: "latest"
+    name: '*'
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml]:
```suggestion
    name: 'rhui-microsoft-azure-rhel*'
```

**Review by to-bar — CHANGES_REQUESTED** [review]

File: ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml (hunk 3 above)
Comment (erzetpe) [reply ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml]: Done.

File: ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml (hunk 4 above)
Comment (erzetpe) [reply ansible/playbooks/roles/repository/tasks/RedHat/install-packages.yml]: Doesn't work properly... so not done. :)

**Review by to-bar — APPROVED** [review]

**Review by atsikham — APPROVED** [review]

**Review by przemyslavic — APPROVED** [review]

**Pull request merged**